File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int ret = 0;
        int row = board.length;
        Stack<Integer> stk = new Stack<>();
        for(int i = 0; i < board.length; i++){
            for(int j = 0; j < board[i].length; j++){
                System.out.print(board[i][j] + " ");
            }
            System.out.println();
        }
        
        for(int i = 0; i < moves.length; i++){
            int k = moves[i] - 1; // 열번호
            boolean flag = false;
            for(int j = 0; j < row; j++){
                if(flag) continue;
                if(board[j][k] != 0){
                    // System.out.println(board[j][k]);
                    // 안비어있고 가장위가 넣을 수랑 같으면
                    if(!stk.isEmpty() && stk.peek() == board[j][k]){
                        // System.out.println(stk.peek() + "::" + board[j][k]);
                        stk.pop();
                        ret++;
                    } else stk.push(board[j][k]);
                    board[j][k] = 0;
                    flag = true;
                }
            }
        }
        // while(!stk.isEmpty()) System.out.println(stk.pop());
        for(int i = 0; i < board.length; i++){
            for(int j = 0; j < board[i].length; j++){
                System.out.print(board[i][j] + " ");
            }
            System.out.println();
        }
        return ret*2;
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    baguni = []
    row, col = len(board), len(board[0])
    ret = 0
    for m in moves:
        pos = m - 1
        for j in range(row):
            if board[j][pos] != 0:
                if baguni and baguni[-1] == board[j][pos]:
                    baguni.pop()
                    ret += 1
                else:
                    baguni.append(board[j][pos])
                board[j][pos] = 0
                break
    return ret * 2
            


"""
인형 없는 곳에서 크레인 작동 안줄거임
터트려져 사라진 인형의 갯수를 리턴

1. move의 위치에서 인형뽑기
2. 바구니 폭발
3. 폭발횟수 * 2 반환
"""